app/src/ui/toolbar/push-pull-button.tsx

```diff
@@ -162,7 +162,10 @@ export class PushPullButton extends React.Component<IPushPullButtonProps, {}> {
       return 'Publish this repository to GitHub'
     }
     if (!this.props.aheadBehind) {
-      return 'Publish this branch to GitHub'
+      const isGitHub = this.props.repository.gitHubRepository !== null
```

Comment:
I don't think we should generally compare to `null` strictly since it won't also catch `undefined`. Instead we could do `!!this.props.repository.gitHubRepository` or `this.props.repository.gitHubRepository != null`.